src/classes/Countertop.js

```diff
@@ -43,8 +44,48 @@ class Countertop {
 		return station
 	}
 
-	updateTopology = () => {
+	/**
+	 * Start the countertop and begin data processing.
+	 *
+	 * This will regenerate the countertop topology to reflect all currently added appliances.
+	 *
+	 * @return {Boolean} Whether the countertop successfully started.
+	 */
+	start = async () => {
+		this.logger.debug('CountertopCoordinator: start()')
+		await this.updateTopology()
+		const promises = this.stations.map(async (station) => station.start())
+		return (await Promise.all(promises)).every((result) => result)
+	}
+
+	/**
+	 * Stop the countertop and halt all data processing.
+	 *
+	 * @return {Boolean} Whether the countertop successfully stopped.
+	 */
+	stop = async () => {
+		this.logger.debug('CountertopCoordinator: stop()')
+		this.stations.map((station) => station.stop())
+		const promises = this.stations.map(async (station) => station.stop())
+		return (await Promise.all(promises)).every((result) => result)
+	}
+
+	/**
+	 * Generates a new topology based on the current list of stations and registers that topology
+	 * with those stations.
+	 *
+	 * Running this is destructive: it will stop processing, remove all previous streams from
+	 * the countertop and replace them with updated streams.
+	 *
+	 * @return {CountertopTopology} The new countertop topology.
+	 */
+	updateTopology = async () => {
+		this.logger.debug('CountertopCoordinator: updateTopology()')
+		await this.stop()
```

Comment:
I think instead of invoking stop, updateTopology should verify that internal state preconditions are met and fail if they are not.

For one thing, this would mean updateTopology can be synchronous, but more importantly it would make the impact of calling it much more predictable.